src/StoreApi/Schemas/ExtendSchema.php

```diff
@@ -67,6 +74,71 @@ public function __construct( Formatters $formatters ) {
 		$this->formatters = $formatters;
 	}
 
+	/**
+	 * Register a custom data control that may be displayed on cart item.
+	 *
+	 * @param array $args {
+	 *     An array of elements that describe the control.
+	 *
+	 *     @type string $namespace    Required. Plugin namespace.
+	 *     @type string $type         One of "checkbox", "select", or "text".
+	 *     @type string $key          Key of the data. Must be unique in this plugin's namespace.
+	 *     @type string $label        The label to show on the front-end.
+	 *     @type string $defaultValue The initial value.
+	 *     @type array  $options   {
+	 *          The options to show if the type is "select"
+	 *
+	 *          @type string $key The key of the option
+	 *          @type string $label The label to show in the select control, the key will be used if no option is supplied.
+	 *
+	 * @throws \Exception On failure to register.
+	 */
+	public function register_custom_cart_item_control( $args ) {
+		$namespace = $args['namespace'];
```

Comment:
santiize_key?

Reply:
Yep, good idea, but I didn't add things like this or other error checking/hardening as it's just a POC.